### DISCUSSION TITLE
HyperHDR-Remote is going to be removed! (VOTE)

### DISCUSSION THREAD

**Discussion by awawa-dev** [awawa-dev/HyperHDR#856] [Polls]
macOS package doesn't have it, there are JSON API, MQTT, GET requests which offer a way to control HyperHDR... so I think this application can be retired. Or at least remove it from our build/installers but the user can compile it from sources on his own. What do you think?

**Comment by Johnny1982fd** [comment]
Hi @awawa-dev 

i need this cause i have some scripts.

But in V20 with LibreElec Addon it is gone?

Script is executable, but:

LibreELEC:~/.kodi/addons/service.hyperhdr/bin # /storage/.kodi/addons/service.hyperhdr/bin/hyperhdr_beenden.sh
/storage/.kodi/addons/service.hyperhdr/bin/hyperhdr_beenden.sh: line 2: /storage/.kodi/addons/service.hyperhdr/bin/hyperhdr-remote: not found
Sending: {'type': 'action', 'content': 'Notification(HyperHDR Info,Ambilight wurde ausgeschaltet,5000,/storage/pictures/Ambilight_OFF.jpg)'}

I tried to copy hyperhdr-remote from v19, but it doesnt work.

What can i do?

**Reply by awawa-dev** [reply]
Nothing has changed with HyperHDR packaging in the official libreelec addon since v19

**Reply by Johnny1982fd** [reply]
It is not included if you installed via addon.

**Reply by awawa-dev** [reply]
Yes, in v19 it was also not included

**Reply by Johnny1982fd** [reply]
Ah ok, is there a way on HyperHDR to make scripts like this ?

**Reply by Johnny1982fd** [reply]
Im testing a little bit, get this one via SSH:

curl -X POST -H "content-type:application/json" http://192.168.133.114:8090/json-rpc?request{"command":"componentstate","componentstate":{"component":"ALL","state":fals
e}}
curl: (3) nested brace in URL position 84:
http://192.168.133.114:8090/json-rpc?request{command:componentstate,componentstate:{component:ALL,state:false}}

**Reply by awawa-dev** [reply]
Simply use `wget`, this is ordinary URL link (GET action, not POST), not JSON either. Or `curl -G <address>`
You can open it also in the web browser

**Reply by Johnny1982fd** [reply]
Does not work:

LibreELEC:~ # wget http://192.168.133.114:8090/json-rpc?request{"command":"componentstate","componentstate":{"component":"ALL","state":false}}
Connecting to 192.168.133.114:8090 (192.168.133.114:8090)
wget: server returned error: HTTP/1.1 404 Not found !

**Reply by Johnny1982fd** [reply]
Im searching for an alternative with my script, like this:

#!/bin/sh
wget http://192.168.133.114:8090/json-rpc?request{"command":"componentstate","componentstate":{"component":"ALL","state":false}}
sleep 1
kodi-send -a "Notification(HyperHDR Info,Ambilight wurde ausgeschaltet,5000,/storage/pictures/Ambilight_OFF.jpg)"
</dev/null >/dev/null 2>&1 &

**Reply by awawa-dev** [reply]
you copied wrong url address from the API command creator in HyperHDR, it's not urlencoded it should be something like that: json-rpc?request=%22command%22%3A%22componentstate%22%2C%22componentstate%22%3A%7B%22component%22%3A%22ALL%22%2C%22state%22%3Afalse%7D%7D chars like {,",} are forbidden for GET and must be encoded

**Reply by Johnny1982fd** [reply]
curl -G http://192.168.133.114:8090/json-rpc?request=%7B%22command%22:%22componentstate%22,%22componentstate%22:%7B%22component%22:%22ALL%22,%22state%22:false%7D%7D

YES, THIS WORKS! THANK YOU!

Now im trying to import this into my script.

**Reply by Johnny1982fd** [reply]
It works, now i change my other scripts.

For all other users who are interested:

#!/bin/sh
curl -G http://192.168.133.114:8090/json-rpc?request=%7B%22command%22:%22componentstate%22,%22componentstate%22:%7B%22component%22:%22ALL%22,%22state%22:false%7D%7D
sleep 1
kodi-send -a "Notification(HyperHDR Info,Ambilight wurde ausgeschaltet,5000,/storage/pictures/Ambilight_OFF.jpg)"
</dev/null >/dev/null 2>&1 &

Make this executable:

chmod a+x /storage/.kodi/addons/service.hyperhdr/bin/hyperhdr_beenden.sh
